### ISSUE TITLE
`sam deploy` does not warn the user of an invalid samconfig.toml

### ISSUE THREAD

**Issue by benkehoe** [aws/aws-sam-cli#2311]
### Description

With an invalid `samconfig.toml`, `sam deploy` does not warn the user of this and proceeds, including failing because necessary arguments provided in the config file are now missing. With `--guided`, it does warn the user of this.

### Steps to reproduce

Create a syntax error in your samconfig.toml, e.g. a missing start quote
```toml
[default.deploy.parameters]
stack_name=my_stack"
```

1. Run `sam deploy`
2. Run `sam deploy --guided`

### Observed result

1.
```
$ sam deploy --debug
Usage: sam deploy [OPTIONS]
Try 'sam deploy --help' for help.

Error: Missing option '--stack-name', 'sam deploy --guided' can be used to provide and save needed parameters for future deploys.
```

2.
```
$ sam deploy --guided --debug
Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics

Configuring SAM deploy
======================

        Looking for config file [samconfig.toml] :  Found
        Reading default arguments  :  Failure
Sending Telemetry: <snip>
Error: Syntax invalid in samconfig.toml; save values through sam deploy --guided to overwrite file with a valid set of values.
```

### Expected result

The error provided with `--guided` is helpful, and and a similar error should be provided whenever `samconfig.toml` has invalid syntax.

### Additional environment details

1. OS: WSL (Ubuntu 18.04) on Windows 10
2. `sam --version`: SAM CLI, version 1.6.2

**Comment by hoffa** [comment]
Thanks for reporting the issue! Fixed in #2316.

**Comment by mndeveci** [comment]
The fix has been released with [SAM CLI v1.7.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.7.0).
